service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/InboundMessageHandlingTest.java

```diff
@@ -0,0 +1,174 @@
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static uk.nhs.adaptors.gp2gp.ehr.EhrStatusConstants.CONVERSATION_ID;
+
+import javax.jms.Message;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.SneakyThrows;
+import uk.nhs.adaptors.gp2gp.common.service.XPathService;
+import uk.nhs.adaptors.gp2gp.common.task.TaskDispatcher;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+import uk.nhs.adaptors.gp2gp.mhs.InboundMessage;
+import uk.nhs.adaptors.gp2gp.mhs.InboundMessageConsumer;
+import uk.nhs.adaptors.gp2gp.testcontainers.ActiveMQExtension;
+import uk.nhs.adaptors.gp2gp.testcontainers.MongoDBExtension;
+
+@RunWith(SpringRunner.class)
+@ExtendWith({SpringExtension.class, MongoDBExtension.class, ActiveMQExtension.class, MockitoExtension.class})
+@SpringBootTest
+@DirtiesContext
+public class InboundMessageHandlingTest {
+    private static final String CONTINUE_INTERACTION_ID = "COPC_IN000001UK01";
+    private static final String ACTION_PATH = "/Envelope/Header/MessageHeader/Action";
+    private static final String CONVERSATION_ID_PATH = "/Envelope/Header/MessageHeader/ConversationId";
+    private static final String CONTINUE_MESSAGE_PAYLOAD = "Continue Acknowledgement";
+
+    @MockBean
+    private XPathService xPathService;
+    @MockBean
+    private ObjectMapper objectMapper;
+    @MockBean
+    private TaskDispatcher taskDispatcher;
+    @Autowired
+    private InboundMessageConsumer inboundMessageConsumer;
+    @Autowired
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+    @Mock
+    private Message message;
+    @Mock
+    private InboundMessage inboundMessage;
+
+    @Test
+    @SneakyThrows
+    public void When_MessageIsUnreadable_Expect_MessageProcessingToBeAborted() {
+        mockNonJsonMessage();
+
+        inboundMessageConsumer.receive(message);
+
+        verify(message, never()).acknowledge();
+        verifyNoInteractions(taskDispatcher);
+    }
+
+    @Test
+    @SneakyThrows
+    public void When_MessageProcessingFails_Expect_WholeProcessToBeFailed() {
+        mockInboundMessage("unsupportedInteractionId", CONTINUE_MESSAGE_PAYLOAD);
```

Comment:
Ideally, for an invalid interaction id I would expect that we don't attempt to perform any processing on the message at all regardless of ConverationId value.

If we encounter an unknown interaction id that means someone has sent a message to an adaptor which is entirely unrelated to GP2GP.

I think it would be acceptable to raise a bug for this if the fix is difficult or out of scope. Both the impact and likelihood are very low.

Reply:
I've changed the cause of failure in this test, so that it's not unsupported interaction ID. When https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/262 is merged, I'll cause message processing to fail in such scenario.

Reply:
I've realised that #262 already handles this case. Just needs to be merged.